### ISSUE TITLE
Mediation libraries support

### ISSUE THREAD

**Issue by ilazakis** [googleads/swift-package-manager-google-mobile-ads#31]
According to the [docs](https://developers.google.com/admob/ios/quick-start#spm), mediation libraries still require using CocoaPods.

<img width="801" alt="image" src="https://github.com/googleads/swift-package-manager-google-mobile-ads/assets/4664657/5b957790-32b7-4a37-9004-373af811b9f2">

Is this on the roadmap, and does the team have an ETA they can share?

Thanks!

**Comment by malandr2** [comment]
Hi @ilazakis,

Thank you for your inquiry. At this time we do not have an ETA that we can share but it is something we are aware is a need for developers (this post helps).

Thanks

**Comment by malandr2** [comment]
For posterity's sake I've re-opened and marked it as a feature request 👍

**Comment by zhutianren-dely** [comment]
google ads and mediations are the last components managed by CocoaPods in my project, really really hope this request could be considered.

**Comment by malandr2** [comment]
You can also file your request on the [Developer Forum](https://groups.google.com/g/google-admob-ads-sdk).

**Comment by ilazakis** [comment]
Hi @malandr2; just checking if there are any updates on ETAs for this? Thanks in advance!

**Comment by rursache** [comment]
@malandr2 any update on this? using cocoapods in 2024 is a joke as SPM is the official package manager. it couldn't take more than a few hours to just support all mediation libraries in SPM. just add the direct links in `Package.swift`

**Comment by cjwirth** [comment]
I'm not sure what the reasoning is for not doing this, but could it be related [to this issue](https://github.com/googleads/googleads-mobile-ios-mediation/issues/448)? 

tl;dr: The mediation libraries use 4-part version numbers (e.g. `12.2.1.0`) when Apple is requiring that it use 3 (`12.2.1`)? I don't claim to know your development flow, but I imagine changing the versioning scheme and/or tools that generate it shouldn't require too much development burden. 

As mentioned in the other thread, 

> Now GoogleMobileAds requires xcode 15.3 and this dependency blocks it. Please post a fix on that!

So it would be very helpful if we would be able to do this.

**Comment by ilazakis** [comment]
Hi @malandr2, thank you for the ongoing discussion and support on this feature request; just checking in for any updates.

With Cocoapods in [maintenance mode](https://blog.cocoapods.org/CocoaPods-Support-Plans/), and very soon in [read-only mode](https://blog.cocoapods.org/CocoaPods-Specs-Repo/), there's presumably a strong enough reason to prioritize this within the team?

Thanks!

**Comment by cjwirth** [comment]
Well, the good news is that some of the mediation libraries that I'm using have since fixed their version numbers, so it's possible to use [the workaround](https://github.com/googleads/googleads-mobile-ios-mediation/issues/413#issuecomment-1532745124) nowadays to upload the frameworks somewhere that you have access to and do it.

Or heck, you could go into the frameworks, update the version numbers (or whatever broken data is in there, as I've found...), re-codesign it, and use that.

Still, I think it's a darn shame that there isn't any official support or communication on this despite CocoaPods literally declaring itself on life support and will be done soon.

**Comment by foreza** [comment]
@cjwirth and all -

We have rolled out SPM support over the last few months for our mediation adapters. 
We now consider SPM the preferred method of integration.

For instance, see https://developers.google.com/admob/ios/choose-networks, and try to add a couple networks of your choice!
